core/src/main/scala/org/http4s/internal/package.scala

```diff
@@ -117,4 +119,47 @@ package object internal {
           }
       }
     }
+
+  /* Derived from https://github.com/typelevel/cats-effect/blob/c62c6c76b3066c500fca2f9e0897605bc12eb2a0/core/shared/src/main/scala/cats/effect/Resource.scala */
+  private[http4s] def interpretResource[F[_], A](resource: Resource[F, A])(
```

Comment:
This is a copy of `Client.toHttpApp`, which should be expressed in terms of this.  But let's agree on whether it should return `ExitCase[Throwable] => F[Unit]` or just `F[Unit]` first.